### PR TITLE
Configure eslint to use Preact linter rules

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We've been using airbnb's linter configuration since the beginning, but it's quite rigid. Since it's release, preact has finally decided on their own linting rules via the https://github.com/preactjs/eslint-config-preact project.

So now we are based off of eslint's recommended configuration followed by Preact's.

One of the things the Preact linter rules gives us as well is now the ability to use `class` instead of `className` for CSS classes in a Preact component.

## QA Instructions, Screenshots, Recordings

N/A

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Actress Lisa Kudrow singing David Bowie's song Changes

### CODE REVIEW

File: app/javascript/.eslintrc.js [hunk 1]
```diff
@@ -2,9 +2,14 @@ const path = require('path');
...
-  extends: ['airbnb', 'plugin:jsx-a11y/recommended', 'prettier'],
+  extends: [
+    'eslint:recommended',
+    'preact',
+    'plugin:jsx-a11y/recommended',
+    'prettier',
+  ],
   parserOptions: {
-    ecmaVersion: 2017,
+    ecmaVersion: 2018,
```
Comment (nickytonline) [inline app/javascript/.eslintrc.js:12]: Updated this to be on par with our other eslint config file.

File: app/assets/javascripts/.eslintrc.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  extends: ['airbnb-base/legacy', 'prettier'],
+  extends: ['eslint:recommended', 'prettier'],
```
Comment (nickytonline) [inline app/assets/javascripts/.eslintrc.js:2]: Using the eslint recommended config now.

File: app/javascript/.eslintrc.js [hunk 3]
```diff
@@ -2,9 +2,14 @@ const path = require('path');
 
 module.exports = {
   parser: 'babel-eslint',
-  extends: ['airbnb', 'plugin:jsx-a11y/recommended', 'prettier'],
+  extends: [
+    'eslint:recommended',
```
Comment (nickytonline) [inline app/javascript/.eslintrc.js:6]: Using the eslint recommended config now along with the Preact eslint recommended configuration.

File: app/javascript/.eslintrc.js [hunk 4]
```diff
@@ -2,9 +2,17 @@ const path = require('path');
...
+    'preact',
+    'plugin:jsx-a11y/recommended',
+    'prettier',
+  ],
   parserOptions: {
-    ecmaVersion: 2017,
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true,
```
Comment (nickytonline) [inline app/javascript/.eslintrc.js:14]: Some config required as we are no longer using the airbnb eslint plugin.

File: app/javascript/.eslintrc.js [hunk 5]
```diff
@@ -28,14 +36,8 @@ module.exports = {
     jest: true,
     browser: true,
   },
-  plugins: ['import', 'jsx-a11y'],
+  plugins: ['import', 'react', 'jsx-a11y'],
```
Comment (nickytonline) [inline app/javascript/.eslintrc.js:39]: Some config required as we are no longer using the airbnb eslint plugin.

**Comment by nickytonline** [comment]
The code climate issue is not related to the PR.

File: app/javascript/actionsPanel/actionsPanel.js [hunk 6]
```diff
@@ -201,7 +201,7 @@ async function adjustTag(el) {
         el.remove();
       } else {
         adjustedTagName = el.value;
-        // eslint-disable-next-line no-param-reassign
+        // eslint-disable-next-line no-param-reassign, require-atomic-updates
```
Comment (nickytonline) [inline app/javascript/actionsPanel/actionsPanel.js:204]: Fixes a code climate issue.

**Comment by rhymes** [comment]
yes!

**Pull request merged**